app_user.go

```diff
@@ -21,59 +30,344 @@ func newAppUser(app App) *appUser {
 // Get gets a user using their userID
 // returns user on success, error on failure
 func (a *appUser) Get(userID string) (*PassageUser, error) {
-	return a.app.GetUser(userID)
+	message := "failed to get Passage User"
+	res, err := a.client.GetUserWithResponse(context.Background(), a.appID, userID)
+	if err != nil {
+		return nil, networkPassageError(message)
+	}
+
+	if res.JSON200 != nil {
+		return &res.JSON200.User, nil
+	}
+
+	var errorCode string
+	switch {
+	case res.JSON401 != nil:
+		errorCode = string(res.JSON401.Code)
+	case res.JSON404 != nil:
+		errorCode = string(res.JSON404.Code)
+		message = fmt.Sprintf(UserIDDoesNotExist, userID)
+	case res.JSON500 != nil:
+		errorCode = string(res.JSON500.Code)
+	}
+
+	return nil, PassageError{
```

Comment:
It was made for the new error. I can make your suggested change for reducing duplication.

Reply:
I attempted the refactor and it seems like it will be harder to avoid all this duplicate code. The original methods return `error` the build in go interface and not `Error`, the passage error object. In order to do the conversion, we would have to do some string manipulation from `error.Error()`. It seems cleaner to remove the duplicate code than to continue down this path.

Reply:
Would a type coercion solve the issue?